rdr_service/genomic/genomic_job_components.py

```diff
@@ -2185,66 +2185,76 @@ def process_samples_into_manifest(self, samples_meta, cohort, saliva=False, loca
 
-            new_array_member_obj = GenomicSetMember(
-                biobankId=bid,
-                genomicSetId=new_genomic_set.id,
-                participantId=samples_meta.pids[i],
-                nyFlag=_ny_flag,
-                sexAtBirth=samples_meta.sabs[i],
-                collectionTubeId=samples_meta.sample_ids[i],
-                validationStatus=(GenomicSetMemberStatus.INVALID if len(valid_flags) > 0
-                                  else GenomicSetMemberStatus.VALID),
-                validationFlags=valid_flags,
-                ai_an='N' if samples_meta.valid_ai_ans[i] else 'Y',
-                genomeType=self._ARRAY_GENOME_TYPE,
-                genomicWorkflowState=GenomicWorkflowState.AW0_READY
-            )
-            # Also create a WGS member
-            new_wgs_member_obj = deepcopy(new_array_member_obj)
-            new_wgs_member_obj.genomeType = self._WGS_GENOME_TYPE
+                # default ny flag if no state id
+                elif not samples_meta.state_ids[i]:
+                    _ny_flag = 0
 
-            inserted_array_member = self.member_dao.insert(new_array_member_obj)
-            inserted_wgs_member = self.member_dao.insert(new_wgs_member_obj)
+                else:
+                    logging.warning(f'No collection site or mail kit state. Skipping biobank_id: {bid}')
+                    continue
 
-            # Add member to PDR
-            for mid in (inserted_array_member.id, inserted_wgs_member.id):
-                bq_genomic_set_member_update(mid, project_id=self.controller.bq_project_id)
-                genomic_set_member_update(mid)
+                new_array_member_obj = GenomicSetMember(
+                    biobankId=bid,
+                    genomicSetId=new_genomic_set.id,
+                    participantId=samples_meta.pids[i],
+                    nyFlag=_ny_flag,
+                    sexAtBirth=samples_meta.sabs[i],
+                    collectionTubeId=samples_meta.sample_ids[i],
+                    validationStatus=(GenomicSetMemberStatus.INVALID if len(valid_flags) > 0
+                                      else GenomicSetMemberStatus.VALID),
+                    validationFlags=valid_flags,
+                    ai_an='N' if samples_meta.valid_ai_ans[i] else 'Y',
+                    genomeType=self._ARRAY_GENOME_TYPE,
+                    genomicWorkflowState=GenomicWorkflowState.AW0_READY
+                )
+                # Also create a WGS member
+                new_wgs_member_obj = deepcopy(new_array_member_obj)
+                new_wgs_member_obj.genomeType = self._WGS_GENOME_TYPE
+
+                processed_array_wgs.extend([new_array_member_obj, new_wgs_member_obj])
+                count += 1
+
+                if count % 1000 == 0:
+                    session.bulk_save_objects(processed_array_wgs)
+                    bulk_ids = self.member_dao.get_members_from_set_id(new_genomic_set.id)
```

Comment:
I understand the need to load the ids for the batch of genomic members that were just inserted, but would this also load the ids from all previously saved batches for this genomic set?

I'm thinking the first batch saves and the ids would load as expected. But then the second batch would load all the ids from the first and second batches (and send re-send the ids from the first batch to the bq update methods).

Reply:
You are correct. Logic has been updated to reflect.